test/PlaidClientTest.js

```diff
@@ -547,51 +547,62 @@ describe('plaid.Client', () => {
         });
       });
 
-      it('transactions', cb => {
-        async.waterfall([
-          cb => {
-            pCl.createItem({
-              username: testConstants.USERNAME,
-              password: testConstants.PASSWORDS.GOOD
-            }, testConstants.INSTITUTION, testConstants.PRODUCTS, {
-              transactions: {
-                start_date: now,
-                end_date: now,
-                await_results: true
-              }
-            },
-            (err, mfaResponse, successResponse) => {
-              expect(err).to.be(null);
-              expect(successResponse).to.be.ok();
-
-              cb(null, successResponse.access_token);
-            });
-          }, (accessToken, cb) => {
-            pCl.getTransactions(accessToken, now, now, {},
-            (err, successResponse) => {
-              expect(err).to.be(null);
-              expect(successResponse).to.be.ok();
-              expect(successResponse.transactions).to.be.an(Array);
+      describe('transactions', () => {
+        let accessToken;
+        beforeEach(done => {
+          pCl.createItem({
+            username: testConstants.USERNAME,
+            password: testConstants.PASSWORDS.GOOD
+          }, testConstants.INSTITUTION, testConstants.PRODUCTS, {
+            transactions: {
+              start_date: now,
+              end_date: now,
+              await_results: true
+            }
+          },
+          (err, mfaResponse, successResponse) => {
+            expect(err).to.be(null);
+            expect(successResponse).to.be.ok();
 
-              cb();
-            });
-          }
-        ], cb);
-      });
+            accessToken = successResponse.access_token;
+            done();
+          });
+        });
 
-      it('transactions (w/o options arg) (with 400)', cb => {
-        pCl.getTransactions('invalid token', now, now,
-        (err, successResponse) => {
-          expect(err).to.be.ok();
-          expect(successResponse).not.to.be.ok();
-          expect(err.status_code).to.be(400);
-          expect(err.request_id).to.be.ok();
-          expect(err.error_code).to.be('INVALID_ACCESS_TOKEN');
+        it('normal flow', cb => {
+          pCl.getTransactions(accessToken, now, now, {},
+          (err, successResponse) => {
+            expect(err).to.be(null);
+            expect(successResponse).to.be.ok();
+            expect(successResponse.transactions).to.be.an(Array);
 
-          cb();
+            cb();
+          });
+        });
+
+        it('all transactions', cb => {
+          pCl.getAllTransactions(accessToken, now, now,
```

Comment:
this is good for an end-to-end test, but I'm not sure it tests anything other than that the function doesn't throw an exception. This is generally OK for the other functions, since they don't do much other than call the API, but now that we are implementing some real logic, I think we need more.

@evanlimanto could we add some unit tests for the pagination? I would suggest using `sinon` to stub out the implementation of `self.getTransactions` to return fake data in pages (and assert that the aggregate result is returned).

Reply:
added unit tests!